pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/DocumentCustomerPortalUrlServlet.java

```diff
@@ -0,0 +1,51 @@
+package com.redhat.pantheon.servlet;
+
+import com.redhat.pantheon.extension.url.CustomerPortalUrlUuidProvider;
+import com.redhat.pantheon.extension.url.UrlProvider;
+import com.redhat.pantheon.model.ModelException;
+import com.redhat.pantheon.model.document.Document;
+import com.redhat.pantheon.model.document.DocumentVariant;
+import com.redhat.pantheon.servlet.util.ServletHelper;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.apache.sling.servlets.annotations.SlingServletResourceTypes;
+import org.jetbrains.annotations.NotNull;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Component;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import java.io.IOException;
+
+@Component(
+        service = Servlet.class,
+        property = {
+                Constants.SERVICE_DESCRIPTION + "=Servlet which provides a Customer Portal URL for any document",
+                Constants.SERVICE_VENDOR + "=Red Hat Content Tooling team"
+        })
+@SlingServletResourceTypes(
+        resourceTypes = { "pantheon/moduleVariant", "pantheon/assemblyVariant", "pantheon/module", "pantheon/assembly" },
+        methods = "GET",
+        selectors = "url",
+        extensions = "txt")
+public class DocumentCustomerPortalUrlServlet extends SlingSafeMethodsServlet {
```

Comment:
I would try to use more generic names than "Customer Portal". These specific classes should eventually live in a different bundle.

Reply:
I partially agree. Someday, these should live in a redhat-specific bundle, and I tried to split the code between "generic definition" and "specific implementation" as much as I could (although this is a servlet so it doesn't apply here). But because that is the ultimate goal, I think it's ok to use specific names in these and other classes that are destined to get broken out from the main codebase. What do you think?

Reply:
Yes, we're not going to split them up now, so as long as we are aware of the future plan I'm good. Perhaps a comment describing the purpose of the class would help remind us of that.